bitnami/argo-workflows/values.yaml

```diff
@@ -672,10 +672,9 @@ controller:
           name: {{ include "argo-workflows.controller.database.password.secret" . }}
           key: {{ include "argo-workflows.controller.database.password.secret.key" . }}
     {{- end }}
-    ## Uncomment to set workflowDefaults
-    ## Ref: https://argoproj.github.io/argo-workflows/default-workflow-specs/#setting-default-workflow-values
-    ## workflowDefaults: {}
-    ##
+    {{- if .Values.controller.workflowDefaults.enabled }}
+    workflowDefaults: {{- include "common.tplvalues.render" (dict "value" .Values.controller.workflowDefaults "context" $) | nindent 2 }}
```

Comment:
I'm afraid this is not as easy as with `telemetryConfig`. It turns out that `telemetryConfig.enabled` is a valid entry:

https://github.com/argoproj/argo-workflows/blob/1bd841853633ebb71fc569b2975def90afb1a68d/docs/workflow-controller-configmap.yaml#L259-L263

Nevertheless, I don't think `workflowDefaults.enabled` would be valid...

https://github.com/argoproj/argo-workflows/blob/1bd841853633ebb71fc569b2975def90afb1a68d/docs/workflow-controller-configmap.yaml#L320-L329

 You may need to add a subentry to `workflowDefaults`, say `workflowDefaults.spec`, where the config should be specified.

Reply:
I added `workflowDefaults.enabled` as a new variable and in my testing it did not introduce any issues, even though it is not a part of the `workflowDefaults` accepted keys. I understand why it might not be desirable to do it.

My new idea then is to remove `enabled` key from `workflowDefaults`, set the default value for `workflowDefaults` as `{}`, then check that `workflowDefaults` dict is not empty when including the contents of it in the configMap


```diff
-  ## @param controller.workflowDefaults.enabled Enable Default Workflow Values
-  workflowDefaults:
-    enabled: false
+  ## @param controller.workflowDefaults Default Workflow Values
+  workflowDefaults: {}


- {{- if .Values.controller.workflowDefaults.enabled }}
+ {{- if .Values.controller.workflowDefaults }}
```

By default this does not add anything to the configMap (same as before).

If user would like to provide Workflow Defaults, they would do the following:
```
workflowDefaults:
  metadata:
    annotations:
      argo: workflows
    labels:
      foo: bar
  spec:
    ttlStrategy:
      secondsAfterSuccess: 5
    parallelism: 3
```

Importantly, the values of `workflowDefaults` are not using multi-line literal style scalar as they do in [linked docs](https://argoproj.github.io/argo-workflows/default-workflow-specs/#setting-default-workflow-values), but "normal" yaml syntax.

Is this option looking better?

Reply:
> My new idea then is to remove enabled key from workflowDefaults, set the default value for workflowDefaults as {}, then check that workflowDefaults dict is not empty when including the contents of it in the configMap

Super nice! 🚀 

> but "normal" yaml syntax.

I reckon this totally makes sense. I will mark this thread as resolved :)